common/app/targeting/CampaignAgent.scala

```diff
@@ -0,0 +1,27 @@
+package targeting
+
+import common._
+import com.gu.targeting.client.CampaignCache
+import conf.Configuration
+import scala.util.control.NonFatal
+import scala.concurrent.Future
+
+object CampaignAgent extends Logging with ExecutionContexts {
+  private val agent = AkkaAgent[CampaignCache](CampaignCache(List()))
+
+  def refresh(): Future[Unit] = {
+    Configuration.targeting.campaignsUrl.map(url => {
+      CampaignCache.fetch(url).map(agent.send)
+    }).getOrElse(Future.failed(throw new BadConfigurationException("Campaigns URL not configured")))
```

Comment:
yep only either remove the word `throw` as you just want to put the exception in as a fail,not throw the exception from the code, or alternatively you could remove the `Future.failed` and let something else catch it.  The former is probably more functional programming compliant but I *think* the scheduler should cope with the latter too